modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java

```diff
@@ -0,0 +1,500 @@
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.workflow.handler.composer;
+
+import org.opencastproject.composer.api.ComposerService;
+import org.opencastproject.composer.api.EncoderException;
+import org.opencastproject.composer.api.EncodingProfile;
+import org.opencastproject.job.api.Job;
+import org.opencastproject.job.api.JobContext;
+import org.opencastproject.mediapackage.MediaPackage;
+import org.opencastproject.mediapackage.MediaPackageElementFlavor;
+import org.opencastproject.mediapackage.MediaPackageElementParser;
+import org.opencastproject.mediapackage.MediaPackageException;
+import org.opencastproject.mediapackage.Track;
+import org.opencastproject.util.NotFoundException;
+import org.opencastproject.workflow.api.AbstractWorkflowOperationHandler;
+import org.opencastproject.workflow.api.WorkflowInstance;
+import org.opencastproject.workflow.api.WorkflowOperationException;
+import org.opencastproject.workflow.api.WorkflowOperationResult;
+import org.opencastproject.workflow.api.WorkflowOperationTagUtil;
+import org.opencastproject.workspace.api.Workspace;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class SelectTracksWorkflowOperationHandler extends AbstractWorkflowOperationHandler {
+  private static final Logger logger = LoggerFactory.getLogger(SelectTracksWorkflowOperationHandler.class);
+
+  /** Name of the 'encode to video only work copy' encoding profile */
+  private static final String PREPARE_VIDEO_ONLY_PROFILE = "video-only.work";
+
+  /** Name of the muxing encoding profile */
+  private static final String MUX_AV_PROFILE = "mux-av.work";
+
+  /** The composer service */
+  private ComposerService composerService = null;
+
+  /** The local workspace */
+  private Workspace workspace = null;
+
+  /** The configuration options for this handler */
+  private static final SortedMap<String, String> CONFIG_OPTIONS;
+
+  private enum AudioMuxing {
+    NONE("none"), FORCE("force"), DUPLICATE("duplicate");
+
+    private final String value;
+
+    AudioMuxing(final String value) {
+      this.value = value;
+    }
+
+    static AudioMuxing fromConfigurationString(final String s) {
```

Comment:
Isn't this basically what the `enum` does anyway and you could e.g. just use `AudioMuxing.valueOf(…)` instead of this custom code?